Sources/TuistSupport/Extensions/String+Extras.swift

```diff
@@ -116,6 +116,7 @@ extension String {
         }
 
         let parts = components(separatedBy: CharacterSet.alphanumerics.inverted)
+            .map { $0.isEmpty ? "_" : $0 }
```

Comment:
Seeing this, maybe the problem was that the `_` was not part of the camelized string initially, I don't think `camelized` should add the `_` itself 🤔

Reply:
The `components` method returns an empty string only at the beginning or end of the string. The issue arises because `camelized` removes characters from both the start and end of the string, which it shouldn't do. This is because special characters at the beginning do not serve the purpose of separating words; rather, they play a more functional role. Moreover, in Swift, names in camel case and special characters at the beginning are standard. For instance, property wrapper names begin with '_' and '$'.

Reply:
It would probably be better to keep at the beginning of the line those characters that were there and not replace them with '_', but this will greatly complicate the implementation and is unlikely to give advantages

Reply:
I had a second look and I now understand this better, think it's fine to keep what you have done